components/shopify/sources/updated-customer/updated-customer.js

```diff
@@ -0,0 +1,36 @@
+const shopify = require("../../shopify.app.js");
+
+module.exports = {
+  key: "shopify-updated-customer",
+  name: "Updated Customer",
+  description: "Emits an event each time a customer's information is updated.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    shopify,
+  },
+  async run() {
+    const lastUpdatedAt = this.db.get("last_updated_at") || null;
+    let results = await this.shopify.getCustomers(null, lastUpdatedAt);
+
+    for (const customer of results) {
+      const dedupeId = `${customer.id}-${customer.updated_at}`;
+      this.$emit(customer, {
+        id: dedupeId,
+        summary: `${customer.first_name} ${customer.last_name}`,
```

Comment:
`<opinion>`
First and last names are personal information that might cause issues with users that require to be compliant with something like GDPR. I'm not an expert in this field so take my opinion with a grain of salt, but perhaps we should log something like `Customer updated: 1234` where `1234` is the customer ID?
`</opinion>`